Cabal/src/Distribution/Simple/Configure.hs

```diff
@@ -988,7 +998,11 @@ dependencySatisfiable
                           -- cabal-testsuite/PackageTests/ConfigureComponent/SubLib/setup-explicit.test.hs
                           || pkgName (IPI.sourcePackageId ipi) == pn)
                     maybeIPI
+                  -- Don't check if it's visible, we promise to build it before we need it.
+                  || promised
       where maybeIPI = Map.lookup (depName, CLibName lib) requiredDepsMap
+            -- Promised is just the same as internal here
+            promised = isJust $ Map.lookup (depName, CLibName lib) promisedDeps
```

Comment:
I am not sure what you mean exactly, `visible`  seems to take into account variety of things which are not to do with multiple public libraries.

This seems the the right place to me and in all our testing didn't cause any issues yet.

Reply:
Thanks for removing the confusing part. And since it's documented that --promised-dependency removes some checks I'll answer my other question positively.